specs/date-picker/date-picker.md

```diff
@@ -0,0 +1,214 @@
+
+- [ ] **Checkpoint 3** - Submit the component PR and schedule a component code review meeting or present the component in one of our Engineering Open Mic sessions. Address feedback from the review, obtain final approvals through GitHub, and merge.
+
+## Overview
+
+As defined by the W3C:
+>The date state represents a control for setting the element's value to a string representing a specific date.
+>
+
+### Background
+
+This is a community requested component.
+
+### Use Cases
+
+- John is booking a room online and needs to enter a date of arrival. Clicking on the calendar icon in the date field, brings up a calendar where John can select a check-in date.
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+  
+### Features
+
+- **Date validation:** Verifies that the entered string is a valid date.
+
+- **Calendar view:** Opens a calendar view for selecting a date.
+
+- **Localization:** Formats that date for the local market. MM--DD-YYYY, DD--MM-YYYY, etc.
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes in order to achieve this component's goals?*
+
+
+
+### Prior Art/Examples
+- [W3C](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/datepicker-dialog.html)
+- [FluentUI](https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker)
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+*Component Name*
+- `fast-date-picker`
+
+*Props/Attrs*
+- `value` - A predefined date or one picked by the user.
+= `locale` - Market and language
+- `placeholder` - Text to appear when a date isn't present.
+- `name` - the name of the control
+- `appearance` - enum
+    - outline (default)
+    - filled
+- `readonly` - If the date is editable.
+- 'disabled` - If the control is disabled.
+- `required` - The date field is required or optional.
+- `min` - Earliest allowed date.
+- `max` - Latest allowed date.cx
+- `allowTextInput` - Whether the date-picker allows input of a date string directly or not.
+- `ariaLabel` - Aria label for screen reader.
+- `type` - Type of picker. Supports all of the HTML types. `date` | `month` | `week` | `time` | `datetime-local`
+
+*Methods*
+
+
+*Events*
+
+- `change` - callback fired when the entered date has changed.
+
+### Anatomy 
+
+**Structure:**
+
+```html
+<div class="date-picker">
+  <div class="label"></div>
+  <div class="date-input">
+    <input type="text" placeholder="${x => x.placeholder}" value=${x => x.value} />
+    <span class="calendar-icon"
```

Comment:
We just switched the 'clear' button to a native html button in the search component. Hopefully we can follow the same pattern here for consistency of styles. I see I may be making an assumption about the difference between clicking the icon and into the text portion. Are we going to show the calendar on default focus into the control, or only on the icon? If only on clicking the icon then I think the pattern defined in search is appropriate here.